src/transformers/models/maskformer/configuration_maskformer.py

```diff
@@ -0,0 +1,179 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" MaskFormer model configuration"""
+import copy
+from typing import Dict, Optional
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+from ..auto.configuration_auto import AutoConfig
+from ..detr import DetrConfig
+from ..swin import SwinConfig
+
+
+MASKFORMER_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "Francesco/maskformer-swin-base-ade": "https://huggingface.co/Francesco/maskformer-swin-base-ade/blob/main/config.json"
+    # See all MaskFormer models at https://huggingface.co/models?filter=maskformer
+}
+
+logger = logging.get_logger(__name__)
+
+
+class MaskFormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`MaskFormer`]. It is used to instantiate a
+    MaskFormer model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the
+    "Francesco/maskformer-swin-base-ade" architecture trained on ADE20k-150
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Currently, maskformer supports only Swin backbone.
+
+    Args:
+        mask_feature_size (`int`, *optional*, defaults to 256):
+            The masks' features size, this value will also be used to specify the Feature Pyramid Network featuresc
+            size.
+        no_object_weight (`float`, *optional*, defaults to 0.1):
+            Weight to apply to the null class .
+        use_auxilary_loss (`bool`, *optional*, defaults to `False`):
+            If `true` [`MaskFormerOutput`] will contain.
+        backbone_config (`Dict`, *optional*):
+            The configuration passed to the backbone, if unset, the configuration corresponding to `swin-base` will be
+            used.
+        detr_config (`Dict`, *optional*):
+            The configuration passed to the Detr model, if unset the base config for `detr` will be used.
+        init_std (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        init_xavier_std (`float`, *optional*, defaults to 1):
+            The scaling factor used for the Xavier initialization gain in the HM Attention map module.
+        dice_weight (`float`, *optional*, defaults to 1.0):
+            The weight for the dice loss.
+        cross_entropy_weight (`float`, *optional*, defaults to 1.0):
+            The weight for the cross entropy loss.
+        mask_weight (`float`, *optional*, defaults to 20.0):
+            The weight for the mask loss.
+        num_labels (`int`, *optional*, defaults to 150):
+            The number of labels.
```

Comment:
```suggestion
```
num_labels shouldn't be in the docs, it's inherits from `PretrainedConfig`.

Reply:
how do I default it to `150` then?

Reply:
What's the reason you want it default to 150?

Reply:
Very good catch, this shouldn't be here indeed!

Reply:
removed :) (I wanted to default to 150 so we create an `maskformer-swin-base-ade`)